tests/core/test_docstring_utils.py

```diff
@@ -0,0 +1,237 @@
+    """
+    expected = DocstringInfo(
+        description="Calculate the area of a rectangle.",
+        params={"width": "The width of the rectangle.", "height": "The height of the rectangle."},
+        returns="float: The area of the rectangle.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_multi_line_param_description_with_colon():
+    """
+    Test parsing a docstring where a parameter description spans multiple lines and includes a colon.
+    """
+    docstring = """
+    Add one to the input.
+
+    Args:
+        x: The input value.
+            For example: 4
+    Returns:
+        int: The input value incremented by one.
+    """
+    expected = DocstringInfo(
+        description="Add one to the input.",
+        params={"x": "The input value. For example: 4"},
+        returns="int: The input value incremented by one.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_no_params():
+    """
+    Test parsing a docstring with no parameters.
+    """
+    docstring = """
+    Get the current timestamp.
+
+    Returns:
+        str: The current timestamp in ISO format.
+    """
+    expected = DocstringInfo(
+        description="Get the current timestamp.",
+        params={},
+        returns="str: The current timestamp in ISO format.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_google_style():
+    """
+    Test parsing a Google-style docstring.
+    """
+    docstring = """
+    Calculate the sum of two numbers.
+
+    Args:
+        a (int): The first number.
+        b (int): The second number.
+
+    Returns:
+        int: The sum of a and b.
+    """
+    expected = DocstringInfo(
+        description="Calculate the sum of two numbers.",
+        params={"a": "The first number.", "b": "The second number."},
+        returns="int: The sum of a and b.",
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_malformed_missing_sections():
+    """
+    Test parsing a malformed docstring missing Args and Returns sections.
+    """
+    docstring = """
+    Just a simple description without parameters or return information.
+    """
+    expected = DocstringInfo(
+        description="Just a simple description without parameters or return information.",
+        params={},
+        returns=None,
+    )
+    result = parse_docstring(docstring)
+    assert result == expected
+
+
+def test_parse_docstring_parameters_without_descriptions():
+    """
+    Test parsing a docstring where parameters are listed without descriptions.
+    """
+    docstring = """
+    Process data.
+
+    Args:
+        data:
+        config:
+    Returns:
+        bool: Success status.
+    """
+    expected = DocstringInfo(
+        description="Process data.", params={}, returns="bool: Success status."
```

Comment:
shouldn't params be `{"a": "", "b": ""}` or `{"a": None, "b": None}`?

Reply:
yep, adjusted accordingly